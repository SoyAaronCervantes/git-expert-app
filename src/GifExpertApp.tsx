import React, {useState} from 'react';
import AddCategoryComponent from "./components/AddCategory";
import GifGridComponent from "./components/GifGrid";
import 'animate.css';

const GifExpertApp = ({ value = [] }: { value: string[] }) => {
  
  const [categories, setCategories] = useState( value );

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategoryComponent setCategories={ setCategories }/>
      <hr/>
      <ol>
        {
          categories.map( category => (
            <GifGridComponent
              key={ category }
              category={ category }
            />
          ))
        }
      </ol>
    </>
  );
};

export default GifExpertApp;
