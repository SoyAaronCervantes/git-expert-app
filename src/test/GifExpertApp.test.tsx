import React from 'react';
import { shallow} from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Test <GifExpertApp/>', () => {
  const wrapper = shallow(<GifExpertApp />);

  test('should display correctly', () => {
    expect( wrapper ).toMatchSnapshot();
  })

  test('should display category list', () => {
    const categories = ['Goku', 'Smite'];
    const wrapper = shallow(<GifExpertApp value={categories} />)
    expect( wrapper ).toMatchSnapshot()
    expect( wrapper.find('GifGridComponent').length ).toBe( categories.length );
  });

});