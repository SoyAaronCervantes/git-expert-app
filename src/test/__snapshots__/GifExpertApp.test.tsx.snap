// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifExpertApp/> should display category list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategoryComponent
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGridComponent
      category="Goku"
      key="Goku"
    />
    <GifGridComponent
      category="Smite"
      key="Smite"
    />
  </ol>
</Fragment>
`;

exports[`Test <GifExpertApp/> should display correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategoryComponent
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
